app/src/main/java/com/pvp/app/service/SettingServiceImpl.kt

```diff
@@ -41,4 +48,36 @@ class SettingServiceImpl @Inject constructor(
 
         context.dataStoreSettings.edit { it[setting.key] = value }
     }
+
+    @Composable
+    override fun <T> remember(
```

Comment:
Could be single line. Also very weird that we have UI logic in business logic. If you can think of a way to make this work without `composable` being _here_ in the business layer, then it would be the preferred way. If not, let it be as it is.